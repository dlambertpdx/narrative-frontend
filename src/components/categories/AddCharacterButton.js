import React from 'react';
import PropTypes from 'prop-types';
import EditFormModalCharacter from './EditFormModalCharacter';

const AddCharacterButton = ({ show, hideModal, showModal }) => {
  return (
    <>
      <EditFormModalCharacter show={show} hideModal={hideModal}></EditFormModalCharacter>
      <div>
        <button type='button' onClick={showModal}>Add New</button>
      </div>
    </>
  );
};

AddCharacterButton.propTypes = {
  show: PropTypes.bool,
  hideModal: PropTypes.func,
  showModal: PropTypes.func
};

export default AddCharacterButton;
