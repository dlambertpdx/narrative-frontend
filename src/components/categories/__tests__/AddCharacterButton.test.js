import React from 'react';
import { shallow } from 'enzyme';
import AddCharacterButton from '../AddCharacterButton';

describe('AddCharacterButton component', () => {
  it('renders an add button', () => {
    const wrapper = shallow(<AddCharacterButton handleAddClick={() => true} />);
    expect(wrapper).toMatchSnapshot();
  });
});