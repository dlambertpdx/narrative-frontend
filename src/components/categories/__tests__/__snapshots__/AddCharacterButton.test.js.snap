// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCharacterButton component renders an add button 1`] = `
<Fragment>
  <EditFormModalCharacter />
  <div>
    <button
      type="button"
    >
      Add New
    </button>
  </div>
</Fragment>
`;
